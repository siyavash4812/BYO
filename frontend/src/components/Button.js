import styled from "styled-components";
import PropTypes from "prop-types";
import { FontSize } from "../utils/sizes";
import Colours from "../utils/colours";

const Button = styled.button`
  font-size: ${({ size }) => FontSize[size]};
  background-color: ${Colours.AQUA};
  color: ${Colours.NAVY};
  height: 40px;
`;

Button.propTypes = {
  size: PropTypes.oneOf(["S", "M", "L"]),
};
Button.defaultProps = {
  size: "S",
};

export default Button;
