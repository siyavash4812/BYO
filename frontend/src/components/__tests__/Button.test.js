import React from "react";
import renderer from "react-test-renderer";

import Button from "../Button";

describe("Button", () => {
  it("renders Button with size 1em if size is not specified", () => {
    const tree = renderer.create(<Button />).toJSON();
    expect(tree).toMatchSnapshot();
    expect(tree).toHaveStyleRule("font-size", "1em");
  });
});
