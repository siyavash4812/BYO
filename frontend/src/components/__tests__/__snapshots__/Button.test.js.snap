// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button renders Button with size 1em if size is not specified 1`] = `
.c0 {
  font-size: 1em;
  background-color: #7FDBFF;
  color: #001f3f;
  height: 40px;
}

<button
  className="c0"
  size="S"
/>
`;
